context/ThemeProvider.tsx

```diff
@@ -0,0 +1,53 @@
+'use client';
+
+import { getCookie } from 'cookies-next';
+import React, {
+  createContext,
+  useContext,
+  useState,
+  useEffect,
+} from 'react';
+
+import { IThemeContext, ModeType } from '@/types';
+
+const ThemeContext = createContext<IThemeContext | undefined>(
+  undefined
+);
+
+export const ThemeProvider = ({
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  const [mode, setMode] = useState<ModeType>('light');
```

Comment:
```suggestion
export const ThemeProvider = ({
  children,
  defaultTheme
}: {
  children: React.ReactNode;
  defaultTheme: string;
}) => {
  const [mode, setMode] = useState<ModeType>(defaultTheme as ModeType);
```
This is where we grab our theme passed down in the layout

Reply:
Notice we set our initial state to the "defaultTheme" being sent from the cookies. This keeps everything in sync